src/elements/fields/SignatureField/components/SignatureModal.tsx

```diff
@@ -0,0 +1,375 @@
+                  >
+                    <div
+                      ref={previewRef}
+                      css={{
+                        fontSize: '1.5em',
+                        fontFamily: 'cursive',
+                        color: '#000'
+                      }}
+                      className='previewText'
+                    >
+                      {fullName}
+                    </div>
+                  </div>
+                  {!signatureImgData ? (
+                    'Full Name'
+                  ) : (
+                    <img
+                      src={signatureImgData}
+                      alt='Signature Image'
+                      css={{ maxHeight: '100%' }}
+                    />
+                  )}
+                </div>
+              </div>
+              <div
+                css={{
+                  height: '1px',
+                  display: 'flex',
+                  alignItems: 'center',
+                  justifyContent: 'center',
+                  borderBottom: '1px solid #e9e9e9'
+                }}
+              >
+                <div
+                  css={{
+                    backgroundColor: '#fff',
+                    padding: '10px',
+                    color: '#5e5e5e'
+                  }}
+                >
+                  or
+                </div>
+              </div>
+              <div
+                css={{
+                  paddingTop: '30px'
+                }}
+              >
+                <div
+                  onClick={() => setDrawSignature(true)}
+                  css={{
+                    padding: '20px',
+                    borderRadius: '4px',
+                    border: '1px solid #e9e9e9',
+                    '& h3': {
+                      padding: 0,
+                      margin: 0,
+                      marginBottom: '10px'
+                    },
+                    '& p': {
+                      margin: 0,
+                      padding: 0
+                    },
+                    '&:hover': {
+                      border: '1px solid #5e5e5e',
+                      cursor: 'pointer'
+                    }
+                  }}
+                >
+                  <h3>Draw your signature</h3>
+                  <p>Draw your signature here using your mouse or trackpad.</p>
+                </div>
+              </div>
+            </>
+          )}
+          {drawSignature && (
+            <div
+              css={{
+                display: 'flex',
+                flexDirection: 'column',
+                gap: '15px'
+              }}
+            >
+              <p
+                css={{
+                  margin: 0,
+                  padding: 0
+                }}
+              >
+                Use your cursor or trackpad to draw your signature in the box
+                below.
+              </p>
+              <div
+                css={{ position: 'relative', width: '100%', height: '200px' }}
+              >
```

Comment:
It would be nice to scale the image canvas so that for reasonably sized signature fields, the resulting signature image extends over the whole field undistorted.  I did this on the original signature dialog I put in place last week.  There are limits of course.  For really wide signature fields (on the form canvas), have some max width to scale to so the canvas on the dialog is reasonably sized and signable.